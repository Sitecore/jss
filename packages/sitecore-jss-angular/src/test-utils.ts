/**
 * Remove angular comments and angular-specific bindings
 * @param {string} html
 */
export function cleanHtml(html: string): string {
  return html
    .replace(/<!--[^>]*-->/g, '')
    .replace(/\s*ng-reflect-[^=]*="[^"]*"/g, '')
    .trim();
}
