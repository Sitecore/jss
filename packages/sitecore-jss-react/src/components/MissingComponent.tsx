import PropTypes from 'prop-types';
import React from 'react';

export interface MissingComponentProps {
  rendering?: {
    componentName?: string;
  };
}

export const MissingComponent: React.FC<MissingComponentProps> = (props) => {
  const componentName =
    props.rendering && props.rendering.componentName
      ? props.rendering.componentName
      : 'Unnamed Component';

  console.log(`Component props for unimplemented '${componentName}' component`, props);

  return (
    <div
      style={{
        background: 'darkorange',
        outline: '5px solid orange',
        padding: '10px',
        color: 'white',
        maxWidth: '500px',
      }}
    >
      <h2>{componentName}</h2>
      <p>
        JSS component is missing React implementation. See the developer console for more
        information.
      </p>
    </div>
  );
};

MissingComponent.propTypes = {
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};

MissingComponent.displayName = 'MissingComponent';
