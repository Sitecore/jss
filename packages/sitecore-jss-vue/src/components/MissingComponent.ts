import { h, defineComponent, PropType } from 'vue';

export const MissingComponent = defineComponent({
  props: {
    rendering: {
      type: Object as PropType<{
        componentName?: string;
      }>,
      default() {
        return undefined as {
          componentName?: string;
        };
      },
    },
  },
  render() {
    const { rendering } = this.$props;
    const componentName =
      rendering && rendering.componentName ? rendering.componentName : 'Unnamed Component';

    console.log(`Component props for unimplemented '${componentName}' component`, this.$props);

    return h(
      'div',
      {
        style: {
          background: 'darkorange',
          outline: '5px solid orange',
          padding: '10px',
          color: 'white',
          maxWidth: '500px',
        },
      },
      [
        h('h2', componentName),
        h(
          'p',
          'JSS component is missing Vue implementation. See the developer console for more information.'
        ),
      ]
    );
  },
});
