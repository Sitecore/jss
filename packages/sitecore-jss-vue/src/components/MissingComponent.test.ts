/* eslint-disable @typescript-eslint/no-empty-function */
import { mount } from '@vue/test-utils';
import { MissingComponent } from './MissingComponent';

describe('<MissingComponent />', () => {
  it('should render', () => {
    const logSpy = jest.spyOn(console, 'log');
    logSpy.mockImplementation(() => {});

    const renderedComponent = mount(MissingComponent, {
      context: {
        props: {
          rendering: { componentName: 'missing' },
        },
      },
    });
    expect(renderedComponent.html()).toMatchSnapshot();
    expect(logSpy).toHaveBeenCalledTimes(1);
    logSpy.mockRestore();
  });
});
