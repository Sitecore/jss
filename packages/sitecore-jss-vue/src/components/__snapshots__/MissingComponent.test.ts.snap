// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissingComponent /> should render 1`] = `
<div style="background: darkorange; outline: 5px solid orange; padding: 10px; color: white; max-width: 500px;">
  <h2>
    missing
  </h2>
  <p>
    JSS component is missing Vue implementation. See the developer console for more information.
  </p>
</div>
`;
